app/engine/RowingStatistics.js

```diff
@@ -140,88 +258,130 @@ function createRowingStatistics (config) {
+    if ((intervalTargetDistance > 0 && rower.totalLinearDistanceSinceStart() >= intervalTargetDistance) || (intervalTargetTime > 0 && rower.totalMovingTimeSinceStart() >= intervalTargetTime)) {
+      return true
+    } else {
+      return false
+    }
   }
 
-  function stopTraining () {
-    sessionState = 'stopped'
-    if (rowingPausedTimer)clearInterval(rowingPausedTimer)
+  function measureRecoveryHR () {
+    // This function is called when the rowing session is stopped. postExerciseHR[0] is the last measured excercise HR
+    // Thus postExerciseHR[1] is Recovery HR after 1 min, etc..
+    if (heartrate !== undefined && heartrate > config.userSettings.restingHR && sessionStatus !== 'Rowing') {
+      log.debug(`*** HRR-${postExerciseHR.length}: ${heartrate}`)
+      postExerciseHR.push(heartrate)
+      if ((postExerciseHR.length > 1) && (postExerciseHR.length <= 4)) {
+        // We skip reporting postExerciseHR[0] and only report measuring postExerciseHR[1], postExerciseHR[2], postExerciseHR[3]
+        emitter.emit('HRRecoveryUpdate', postExerciseHR)
+      }
+      if (postExerciseHR.length < 4) {
+        // We haven't got three post-exercise HR measurements yet, let's schedule the next measurement
+        setTimeout(measureRecoveryHR, 60000)
+      }
+    }
   }
 
-  function resetTraining () {
-    stopTraining()
-    distanceTotal = 0.0
-    strokesTotal = 0
-    caloriesTotal = 0.0
-    durationTotal = 0
-    caloriesAveragerMinute.reset()
-    caloriesAveragerHour.reset()
-    strokeAverager.reset()
-    powerAverager.reset()
-    speedAverager.reset()
-    powerRatioAverager.reset()
-    sessionState = 'waitingForStart'
+  function emitWebMetrics () {
+    emitMetrics('webMetricsUpdate')
   }
 
-  // clear the metrics in case the user pauses rowing
-  function pauseRowing () {
-    strokeAverager.reset()
-    powerAverager.reset()
-    speedAverager.reset()
-    powerRatioAverager.reset()
-    lastStrokeState = 'RECOVERY'
-    sessionState = 'paused'
-    emitter.emit('rowingPaused')
+  function emitPeripheralMetrics () {
+    emitMetrics('peripheralMetricsUpdate')
+  }
+
+  function emitMetrics (emitType = 'webMetricsUpdate') {
+    emitter.emit(emitType, getMetrics())
+  }
+
+  function getMetrics () {
+    const cyclePace = cycleLinearVelocity.clean() !== 0 && cycleLinearVelocity.raw() > 0 && sessionStatus === 'Rowing' ? (500.0 / cycleLinearVelocity.clean()) : Infinity
+    return {
+      sessiontype: intervalTargetDistance > 0 ? 'Distance' : (intervalTargetTime > 0 ? 'Time' : 'JustRow'),
+      sessionStatus,
+      strokeState: rower.strokeState(),
+      totalMovingTime: totalMovingTime > 0 ? totalMovingTime : 0,
+      driveLastStartTime: driveLastStartTime > 0 ? driveLastStartTime : 0,
+      totalMovingTimeFormatted: intervalTargetTime > 0 ? secondsToTimeString(Math.round(Math.max(intervalTargetTime - totalMovingTime), 0)) : secondsToTimeString(Math.round(totalMovingTime - intervalPrevAccumulatedTime)),
```

Comment:
`(Math.round(Math.max(intervalTargetTime - totalMovingTime), 0))` this error seems to be back (i.e. that the zero is not within the max function call

Reply:
Thanks, I killed that bug. Seems a bad copy on my dev device.